### PR TITLE
Login

### PR DESCRIPTION
**Pull request by caijiah**
Refactor UserLocalStorage.java: renamed it to UserController.java and change methods to PostMapping.
Implemented UserService for following MVC design pattern.

### CODE REVIEW

File: src/main/java/com/example/wbdvsp21finalprojectyeswr21serverjava/models/User.java [hunk 1]
```diff
@@ -4,6 +4,18 @@
...
     private String password;
 
+    public User() {
+    }
+
+    public User(String username, String password) {
+        this.username = username;
+        this.password = password;
+    }
+    public User(User other) {
```
Comment (yiyahe6project) [inline src/main/java/com/example/wbdvsp21finalprojectyeswr21serverjava/models/User.java:14]: what is this for?

File: src/main/java/com/example/wbdvsp21finalprojectyeswr21serverjava/services/UserService.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.servlet.http.HttpSession;
+
+@Service
+public class UserService {
+    List<User> users = new ArrayList<User>();
+
+    public User register(User user, HttpSession session) {
```
Comment (yiyahe6project) [inline src/main/java/com/example/wbdvsp21finalprojectyeswr21serverjava/services/UserService.java:16]: Thanks for restructuring the java server code. I would expect you to implement the login/register/profile UI which can take user input and store info in the UserService. Currently we only use url to make sure the service is functional.

**Review by yiyahe6project — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/wbdvsp21finalprojectyeswr21serverjava/models/User.java (hunk 1 above)
Comment (caijiah) [reply src/main/java/com/example/wbdvsp21finalprojectyeswr21serverjava/models/User.java:14]: A copy constructor for copying a user, I used this for setting password to "***" in register process.

File: src/main/java/com/example/wbdvsp21finalprojectyeswr21serverjava/services/UserService.java (hunk 2 above)
Comment (caijiah) [reply src/main/java/com/example/wbdvsp21finalprojectyeswr21serverjava/services/UserService.java:16]: UI was implemented in our React Repo.

**Pull request merged**